packages/common/src/codegen/utils/contractToInterface.ts

```diff
@@ -39,13 +50,17 @@ export function contractToInterface(
   symbolImports: SymbolImport[];
   qualifiedSymbols: Map<string, QualifiedSymbol>;
 } {
-  let ast: SourceUnit;
-  try {
-    ast = parse(source);
-  } catch (error) {
-    throw new MUDError(`Failed to parse contract ${contractName}: ${error}`);
+  const parser = Parser.create(LanguageFacts.latestVersion());
```

Comment:
I wonder if it wouldn't be better to use the pragma version
```suggestion
  const parser = Parser.create(LanguageFacts.inferLanguageVersions(source)[-1]);
```

Reply:
I'm not sure this is a good idea. But fixing it to the latest supported version may not be good either. The version of `solc` used in MUD seems to be hard-coded, or dictated by foundry.

Reply:
It seems it is hardcoded to `0.8.24` everywhere, and it is replaced manually when needed. Example:
https://github.com/latticexyz/mud/commit/aabd30767cdda7ce0c32663e7cc483db1b66d967

I suggest hardcoding it here as well for compatibility.